### ISSUE TITLE
Split the Validation module into several modules + a package module

### ISSUE THREAD

**Issue by AndrejMitrovic** [bosagora/agora#704]
Since the Validation module is becoming quite big, it would be useful to split its validation routines into separate modules. For example:

```
src/agora/consensus/validation/Block.d
src/agora/consensus/validation/Transaction.d
src/agora/consensus/validation/Enrollment.d
```

In addition to that, we should add a _package module_:

```
src/agora/consensus/validation/package.d
```

And essentially do:

```D
import agora.consensus.validation.Block;
import agora.consensus.validation.Transaction;

/// merge overload sets
public alias isValid = agora.consensus.validation.Block.isValid;
public alias isValid = agora.consensus.validation.Transaction.isValid;
```

@bpfkorea/core:

- this is what a package module is https://dlang.org/spec/module.html#package-module
- The need to "merge overload sets" is described in bullet point 5 here: https://dlang.org/spec/function.html#overload-sets

**Comment by linked0** [comment]
Tranfering each validation codes into each related classes would be better like by trasfering the `isInvalidEnrollmentReason` function into the `EnrollmentManager`. There is two reasons for that. First, each class know well how to check the validity of an object. Second, a developer have to add validation code into the `Validation.d` whenever new class is defined which need to have a validation code, which is not efficient.

**Comment by AndrejMitrovic** [comment]
Then to test validation of anything you would need to instantiate that class. And that means preparing a lot of state. `EnrollmentManager` was already a big class and got split into two.

We want to move away from having big classes with many methods inside.

**Comment by AndrejMitrovic** [comment]
Also with the UFCS feature in D, you can write `transaction.isValid` or `isValid(transaction)`. So it is the same whether the validation function is inside or outside the structure itself.

**Comment by linked0** [comment]
I am respectful of your opinion. But all classes have their role and checking the validity of an object should be contained in the related class because an class know how to deal with its information better than others. It might not be proper that some functions are contained into a module without defining a class.

**Comment by Geod24** [comment]
The comment on that commit was obviously wrong, it fixes #586 , not this issue.
